### PR TITLE
Birthday

### PR DESCRIPTION
**Pull request by dfuentes**
Initial commit for a tool to replay logs (mostly for heka testing)

### CODE REVIEW

**Comment by dfuentes** [comment]
also cc: @peggyl to see if it works for your metrics work (also for a free golang review :) )

File: main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("error reading input: %s", err)
+	}
+	close(readChan)
+	wg.Wait()
+}
+
+func output(wg *sync.WaitGroup, input <-chan string, speed int) {
+	tickerInterval := time.Millisecond * time.Duration(1000/speed)
```
Comment (peggyl) [inline main.go]: Do we lose lines if speed doesn't divide 1000?

Comment (peggyl) [inline main.go]: Never mind, just takes an extra tick to get the last line. :+1:

**Comment by peggyl** [comment]
LGTM so far; will look for an opportunity to try it out this week.

File: main.go (hunk 1 above)
Comment (dfuentes) [inline main.go]: really we don't need this channel indirection at all.  I only separated it to make it easier to test.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,36 @@
-# log-replay
-Replays log files at a given speed
+# Log-replay
+
+`log-replay` is a tool for replaying a given log at a given speed.  The tool is designed to be CLI friendly, allowing pipes where possible.
```
Comment (mohit) [inline README.md]: :100: command line!

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		go output(&wg, inChan, test.speed)
+		for _, in := range test.input {
+			inChan <- in
+		}
+		close(inChan)
+		wg.Wait()
+		finished := time.Now()
+		expected := start.Add(test.expectedDuration)
+
+		assert.WithinDuration(t, expected, finished, time.Millisecond*10)
```
Comment (mohit) [inline main_test.go]: TIL WithinDuration

**Comment by mohit** [comment]
LGTM!

**Pull request merged**